enclave-runtime/src/top_pool_execution.rs

```diff
@@ -0,0 +1,180 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::{
+	error::{Error, Result},
+	ocall::OcallApi,
+	sidechain_block_composer::BlockComposer,
+	sidechain_impl::{exec_aura_on_slot, ProposerFactory},
+	sync::{EnclaveLock, EnclaveStateRWLock},
+	top_pool_operation_executor::{ExecuteGettersOnTopPool, TopPoolOperationExecutor},
+};
+use itc_light_client::{io::LightClientSeal, BlockNumberOps, LightClientState, NumberFor};
+use itp_extrinsics_factory::ExtrinsicsFactory;
+use itp_nonce_cache::GLOBAL_NONCE_CACHE;
+use itp_settings::sidechain::SLOT_DURATION;
+use itp_sgx_crypto::{AesSeal, Ed25519Seal};
+use itp_sgx_io::SealedIO;
+use itp_stf_executor::executor::StfExecutor;
+use itp_stf_state_handler::{query_shard_state::QueryShardState, GlobalFileStateHandler};
+use itp_types::{Block, H256};
+use its_sidechain::{
+	primitives::types::block::SignedBlock as SignedSidechainBlock,
+	slots::{duration_now, remaining_time, sgx::LastSlotSeal, yield_next_slot},
+	top_pool_rpc_author::{global_author_container::GlobalAuthorContainer, traits::GetAuthor},
+};
+use log::*;
+use sgx_types::sgx_status_t;
+use sp_runtime::traits::Block as BlockT;
+use std::sync::Arc;
+
+#[no_mangle]
+pub unsafe extern "C" fn execute_trusted_getters() -> sgx_status_t {
+	if let Err(e) = execute_top_pool_trusted_getters_on_all_shards() {
+		return e.into()
+	}
+
+	sgx_status_t::SGX_SUCCESS
+}
+
```

Comment:
In this file we have the top-level e-calls for executing trusted getters and trusted calls from the top pool. These will be further refactored and moved to the sidechain crate in the next PR

Reply:
I know it will be further refactored - but what do you think about adding top-level-file description, i.e. documenting the thought process of about what should be placed in this file?

What I'm dreaming about: 
https://github.com/paritytech/substrate/blob/master/frame/scheduler/src/lib.rs#L18-L48

What might be reality: 
https://github.com/paritytech/substrate/blob/master/primitives/core/src/hash.rs#L18

But we could try?